lib/rubocop/cop/performance/detect.rb

```diff
@@ -51,7 +51,7 @@ def autocorrect(node)
         private
 
         def preferred_method
-          cop_config['PreferredMethod']
+          config.for_cop('Style/CollectionMethods')['PreferredMethods']['detect'] # rubocop:disable Metrics/LineLength
```

Comment:
This works with the default configuration where the hash contains `'detect' => 'find'` meaning that `find` is preferred. If the configuration is overridden to make `detect` the preferred choice, then the `'detect'` key will return `nil` and you get `'detect'` from the `'find'` key.

Hope that was clear. :smile: